### ISSUE TITLE
Bug in model assign()

### ISSUE THREAD

**Issue by sszdh** [phalcon/cphalcon#10557]
Hi,

I think, in `assign()` method body [here](https://github.com/phalcon/cphalcon/blob/master/phalcon/mvc/model.zep#L477), setter method should be camelize before call

```php
// Try to find a possible getter
let possibleSetter = "set" . Text::camelize(attributeField);
if method_exists(this, possibleSetter) {
	this->{possibleSetter}(value);
} else {
	let this->{attributeField} = value;
}
```

see the [__get()](https://github.com/phalcon/cphalcon/blob/master/phalcon/mvc/model.zep#L4095)

Thanks

**Comment by stibiumz** [comment]
In PHP functions, methods, classes, traits and interfaces are not case sensitive. `setFoo` is the same as `setfoo` or `sEtFoO` etc

**Comment by sszdh** [comment]
@stibiumz Actually this is not about case sensitiveness, this is about underscore field name like foo_bar, etc.

**Comment by sszdh** [comment]
Actually I think there is an integrity issue in `__set()` and `assign()` functionality.

As I though, the phalcon's rule is to use getter/setters for inaccessible model's properties, but in `assign()` method this rule has been bypassed!

So if we improve `__set()` method as described [here](https://forum.phalconphp.com/discussion/7321/model-set-method-for-array-params), then there is no need to have these 
lines at all:

```php
// Try to find a possible getter
let possibleSetter = "set" . Text::camelize(attributeField);
if method_exists(this, possibleSetter) {
    this->{possibleSetter}(value);
} else {
    let this->{attributeField} = value;
}
```

just using `let this->{attributeField} = value;` and leaving rest job to `__set()` method.

This is my POV.

Sorry my English! :)

Thanks.

**Comment by andresgutierrez** [comment]
Fixed in the 2.0.x branch

**Comment by sszdh** [comment]
Brilliant.
